daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ScenarioRunner.scala

```diff
@@ -451,4 +443,29 @@ object ScenarioRunner {
       // to avoid breaking all tests we keep it for now at least.
       Time.Timestamp.MinValue,
     )
+
+  sealed abstract class ScenarioResult {
+    def ledger: ScenarioLedger
+    def traceLog: TraceLog
+  }
```

Comment:
```suggestion
  } extends Product with Serializable
```
I'm fairly sure @S11001001 wrote about it but I couldn't find his post, [this](https://typelevel.org/blog/2018/05/09/product-with-serializable.html) is what I found. No need to accept this if you don't think it adds anything here.

Reply:
I know about the pattern but I usually add it only when I actually use it in a way where type inference benefits from it. But I guess there is no harm in adding it prematurely.